libbeat/beat/pipeline.go

```diff
@@ -34,6 +34,9 @@ type ClientConfig struct {
 	// Fields provides additional 'global' fields to be added to every event
 	Fields common.MapStr
 
+	// DynamicMeta provides additional fields to be added to every event, supporting live updates
+	DynamicMeta *common.MapStrPointer
```

Comment:
To reduce confusion, rather use `DynamicFields`. The beat event has `Meta` and `Fields`. The `Meta` are somewhat internal (like LS `@metadata`). But `DynamicMeta` is updating the event `Fields`, not `Meta`.

Btw. instead of adding the functionality to the publisher pipeline, you can just pass a processor doing the magic.

Instead of passing DynamicMeta, we could add atomic values support to the normalizer of `common.MapStr`. This would allow you to pass updatable fields via `Fields` (given you don't need them on top-level).

Reply:
Thanks for the input, I agree DynamicFields is the right name.

I thought about the processor but that would cause Copy to be called for all cases, I guess that's not too bad?